ludwig/trainers/trainer_lightgbm.py

```diff
@@ -525,12 +523,19 @@ def train_step(
             eval_set=[(ds.get_data(), ds.get_label()) for ds in eval_sets],
             eval_names=eval_names,
             # add early stopping callback to populate best_iteration
-            callbacks=[lgb.early_stopping(boost_rounds_per_train_step)],
+            callbacks=[lgb.early_stopping(boost_rounds_per_train_step), store_predictions(train_logits)],
             # NOTE: hummingbird does not support categorical features
             # categorical_feature=categorical_features,
         )
         evals_result.update(gbm.evals_result_)
 
+        if not self.evaluate_training_set:
```

Comment:
Should this be `if self.evaluate_training_set`?

Reply:
This is as intended; if `evaluate_training_set == False`, we will use the in-training predictions to compute training set metrics.

Reply:
Ah, got it. If evaluate_training_set == True, we would skip this block, but run evaluation on the entire training set.